### PR TITLE
first commit

### PR DESCRIPTION
**Pull request by lilly-weng**

### CODE REVIEW

File: homeworks/week5/hw1.md [hunk 1]
```diff
@@ -1,3 +1,38 @@
 ## 前四週心得與解題心得
 
+### 前四週心得
 
+Week 1 暖身週
+一開始覺得為什麼暖身週是學習 command line 和 git 這些東西，我想像中的前端應該是 HTML、CSS 啥的，而且看了計畫大綱後才發現這些都要到第五週才開始接觸。學習 command line 和 git 後才發現原來不僅繳交作業都要用到，很多東西都會用到，像是要 run JavaScript 之類的。  
+我覺得老師的課程安排很奇妙，很不同於一些線上課程，一開始就從 command line 和 git 開始教，並且還鼓勵我們要寫心得和筆記，讓我覺得這不只是上課，而是一個習慣的培養，讓我們未來接觸新東西的時候也可以透過心得和筆記溫故知新。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw1.md:7]: 這樣的課程規劃算是老師「精心」設計過的，可以參考老師的部落格文章：[紮實的網頁前端學習路線與資源推薦](https://blog.huli.tw/2019/08/21/real-front-end-learning-path/)，我自已真的覺得很不錯！

File: homeworks/week5/hw1.md [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+
+Week 2 程式基礎 (上) & Week 3 程式基礎 (下)
+這兩週主要就是開始學習 JavaScript 和了解刷題是怎麼一回事，也了解了 node.js 以及 npm 的使用方式。除了偶而需要查詢一些 JavaScript 的語法之外，覺得思考題目的方式是很重要的訓練，需要把一大團的想法，按照順序排列，在一條條的寫成 code，因此要好好思考前後以及限制條件這些，一開始有點痛苦，但有時候也覺得蠻有趣的。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw1.md:10]: 學習的過程總是會遇到難關，但是難關越困難，跨過去之後的成就感也就越大囉！

File: homeworks/week5/hw1.md [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+
+Week 4 網路基礎
+這週除了概念的講解外，主要就是 API 的運用，我覺得 API 是很有趣的東西，可以讓 application 做很多有趣的事情，但是比較苦惱的就是要閱讀 API 文件，而且有時候要取複雜一點的東西就覺得很困難。希望自己能多多熟悉 API 這個部分。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw1.md:13]: API 文件通常面向的讀者就不是一般使用者，剛開始看會很苦惱也是正常的，看多了，套路熟了就會輕鬆很多，其實跟寫 code 或看別人的 code 也是異曲同工啦！

File: homeworks/week5/hw1.md [hunk 4]
```diff
@@ -1,3 +1,38 @@
...
+
+總體來說，偶而會卡住有點痛苦之外，很明顯地看到自己的進步。參加這個 bootcamp 給我的感覺就像是拿到藏寶圖，以前的我就是到處亂走到處碰壁，但是現在手上有藏寶圖的我，知道方向，雖然也會遇到挑戰，可以我知道自己要往哪裡走，所以會全力以赴。我真的很高興自己有參加這個 bootcamp 。 希望接下來的日子也可以繼續加油、繼續學習!
+
+### 解題心得
+
+#### Show time | 公車即時動態
+
+做的時候發現網站的網址有所變更，從網址輸入的不是公車的號碼，而是公車的 ID ，但除此之外其他都是一樣的，不過遇到了一個很奇怪的問題，可以取到站名，但不知道為什麼未發車，或是到站時間這些都沒有抓到，因為也不是很懂，所以就先把影片看完了。老師提到這個影片還有一些要注意的點，就是縮排、變數命名和什麼時候切 function，由於現在都有什麼 prettier 這種工具，所以縮排上我就是比較無腦，倒是變數命名蠻燒腦的，有時候的確不知道要取什麼令人一目了然的名字。最後就是切 function，就算看完影片感覺還是沒有抓到切 function 的感覺，可能就也先從觀察開始慢慢累積經驗後就比較能了解吧。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw1.md:21]: 好像看別的同學的心得有提到公車網頁上的資料顯示方法有變動，資料好像抓不到了？
變數命名、函式規劃……這些同理看 API 文件與寫程式，真的就是看多了，寫多了才會比較有感覺，大概很難速成。

File: homeworks/week5/hw1.md [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+一開始想說要判斷玩家的是奇數還是偶數，後來想一想覺得只要判斷 A 和 B 的個數就好了。於是就先 for 迴圈遍歷所有的人，把 A 和 B 的個數算出來，如果是 A 多就再遍歷一次迴圈，把是 B 的 index 印出來，反之亦然。  
+提交後遇到了錯誤，想想有兩個可能性沒有考慮到，第一個就是如果玩家只有一個人，第二格則是所有人選 A 或是所有人選 B。把這兩個條件加上去之後就沒有問題了。 第一個條件讓我想到老師之前說要注意範圍這件事，我想有些 corner case 都是和最小數或是最大數有關係吧。另外我也覺得解題有時候都會沒有想到一些可能性，像是第二個條件我一開始也沒有想到，由於 test case 看得到的只有一個，因此要想破頭才能想到之前沒有考慮到。
+
+#### LIOJ 1017 貪婪的小偷
+
+一開始我只有考慮到 test case 的狀況，就是要怎麼去取得前幾大有價值的東西，在這邊想用 sort 去排列，因此需要定義一個空的 array 去存放東西，之後再依照包包的容量大小去取前幾位，但提交後出現錯誤訊息。想想後發現自己沒有考慮到包包的容量大於所有的東西的這個情況，這個情況就是要把所有的東西都帶走。
+
+#### LIOJ 1018 大平台
+
+這題我用的是快慢指針，這個可以讓我去紀錄同樣數字的長度，不過到底是什麼時候或是誰加多少讓我想了一下，另外這一題也需要一個 max 的變數來記錄最大的平台有多大這樣。還有也要考慮 corner case 就是只有一個平台的狀況。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw1.md:38]: 其實自已考慮測試資料也是蠻重要的一個訓練，因為之後寫網頁，很多「特殊」狀況都要自已去測試，要把使用者各種奇怪的行為都考慮到網站才不會出錯 XD
感覺你解題過程都有不錯的思考也有得到一些經驗，慢慢累積相信半年之後會有很大的進步的！

File: homeworks/week5/hw2.md [hunk 6]
```diff
@@ -1,3 +1,9 @@
 ## 從歷史中學習
 
+一開始我看學長姐的分享都會不自覺地瞄向求職的部分，因為還是蠻在意之後求職的奇況，不過看到第三期的一個學長姊的分享，其中有一個"學習策略"分享，他主要分成三個部分，第一個是專案優先、第二是廣度導向、第三是時程兼顧。
```
Comment (JAS0NHUANG) [inline homeworks/week5/hw2.md:3]: 這三個方向真的不錯，能確實執行的話應該可以有不錯的學習成果。

**Review by JAS0NHUANG — COMMENTED** [review]
看你目前的學習狀況應該是沒有什麼問題，第六、七、八週就會開始刻出畫面以及簡單的互動網頁，可以更加熟練 JavaScript 的應用，也會更有成就感一些，繼續加油！:smile:
另外，沒有看妳提到 Lidemy HTTP Challenge，如果有空的話也可以試著玩玩看喔，會對 HTTP 請求、回應的認識更具體一些！
